packages/api-client/src/components/Sidebar/Sidebar.vue

```diff
@@ -1,23 +1,179 @@
 <script setup lang="ts">
 import { useWorkspace } from '@/store/workspace'
+import { defineProps, onMounted, ref, watch } from 'vue'
 
+const props = withDefaults(defineProps<{ showSideBar: boolean }>(), {
+  showSideBar: true,
+})
+const emit = defineEmits<{
+  (e: 'update:showSideBar', v: boolean): void
+}>()
 const { isReadOnly } = useWorkspace()
+const isDragging = ref(false)
+const sidebarRef = ref<HTMLElement | null>(null)
+const sidebarWidth = ref(localStorage.getItem('sidebarWidth') || '280px')
+
+/**
+ * Resets the sidebar width to the default value to prevent
+ * having it shrinked on click display after dragging.
+ */
+const initialSidebarWidth = () => {
```

Comment:
Nitpick but would prefer a function declaration here and an action word for function naming such as:

```ts
function setInitialSidebarWidth()
```